WooCommerce/Classes/Authentication/Jetpack Setup/Site Credential Login/SiteCredentialLoginViewModel.swift

```diff
@@ -66,32 +61,64 @@ private extension SiteCredentialLoginViewModel {
             .map { $0.isEmpty || $1.isEmpty }
             .assign(to: &$primaryButtonDisabled)
     }
-}
 
-// MARK: LoginFacadeDelegate conformance
-//
-extension SiteCredentialLoginViewModel: LoginFacadeDelegate {
-    func displayRemoteError(_ error: Error) {
-        isLoggingIn = false
+    func loginAndAttemptFetchingJetpackPluginDetails() {
+        guard let loginURL = URL(string: siteURL + Constants.loginPath),
+              let adminURL = URL(string: siteURL + Constants.adminPath) else {
+            DDLogWarn("⚠️ Cannot construct login URL and admin URL for site \(siteURL)")
+            return
+        }
+
+        // Makes sure the loading indicator is shown
+        isLoggingIn = true
+
+        // Prepares the authenticator with username and password
+        let authenticator = CookieNonceAuthenticator(username: username,
+                                                     password: password,
+                                                     loginURL: loginURL,
+                                                     adminURL: adminURL,
+                                                     version: Constants.defaultWPVersion,
+                                                     nonce: nil)
+        let network = WordPressOrgNetwork(authenticator: authenticator)
+        let authenticationAction = JetpackConnectionAction.authenticate(siteURL: siteURL, network: network)
+        stores.dispatch(authenticationAction)
+
+        // Retrieves Jetpack plugin details to see if the authentication succeeds.
+        let jetpackAction = JetpackConnectionAction.retrieveJetpackPluginDetails { [weak self] result in
+            guard let self else { return }
+            self.isLoggingIn = false
+            switch result {
+            case .success:
+                // Success to get the details means the authentication succeeds.
+                self.handleCompletion()
+            case .failure(let error):
+                self.handleRemoteError(error)
+            }
+        }
+        stores.dispatch(jetpackAction)
+    }
```

Comment:
Super nit: Could we extract this into a method like `func retrieveJetpackPluginDetails()`?